src/primitives/transaction.h

```diff
@@ -159,20 +159,22 @@ class CTxOut
 public:
     CAmount nValue;
     CScript scriptPubKey;
+    bool m_silentpayment;
```

Comment:
In my opinion, this flag doesn't belong here because only the receiver/sender of a silent payment would care about it, which would make it a piece of wallet meta-data.
 
Without knowing much about the wallet my guess would be that it should be stored on `CWalletTx`?